File: javascript-solution/thirdmax.js
/**
https://github.com/bhatiadheeraj/Leetcode-Solution.git

LeetCODE LINK : https://leetcode.com/problems/third-maximum-number/submissions/
**/

var thirdMax = function(nums) {
    let numsSorted = nums.sort((a,b) => b-a).filter((item, index) => nums.indexOf(item) === index);
    if(numsSorted.length >=3) return numsSorted[2];
    return numsSorted[0];
};

File: python_solutions/thirdmax.py
#question :  https://leetcode.com/problems/third-maximum-number/submissions/
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        numSorted = list(set(nums))
        numSorted = (sorted(numSorted, reverse=True))
        if len(numSorted) >= 3: return numSorted[2]
        return numSorted[0]
